app/ide-desktop/lib/performance-logger/src/performance_logger.ts

```diff
@@ -0,0 +1,245 @@
+export enum LogLevel {
+    Section = 'SECTION',
+    Task = 'TASK',
+    Detail = 'DETAIL',
+    Debug = 'DEBUG',
+}
+
+/// Delimiter used to to encode information in the `PerformanceEntry` name.
+const MESSAGE_DELIMITER = '//'
+
+const ENABLE_SECTION_PROFILING = process.env.PROFILING_LEVEL.toLowerCase().includes('section')
+const ENABLE_TASK_PROFILING = process.env.PROFILING_LEVEL.toLowerCase().includes('task')
+const ENABLE_DETAIL_PROFILING = process.env.PROFILING_LEVEL.toLowerCase().includes('detail')
+const ENABLE_DEBUG_PROFILING = process.env.PROFILING_LEVEL.toLowerCase().includes('debug')
+
+/// Check whether the given log level should perform any logging activity.
+function log_level_is_active(log_level: LogLevel): boolean {
+    switch (log_level) {
+        case LogLevel.Section:
+            return ENABLE_SECTION_PROFILING
+        case LogLevel.Task:
+            return ENABLE_TASK_PROFILING
+        case LogLevel.Detail:
+            return ENABLE_DETAIL_PROFILING
+        case LogLevel.Debug:
+            return ENABLE_DEBUG_PROFILING
+    }
+}
+
+/// Return a string that encodes the given log level and name for a mark that indicates the start of
+/// an interval.
+function start_interval_label(log_level: LogLevel, interval_name: string): string {
+    return log_level.toString() + MESSAGE_DELIMITER + interval_name + MESSAGE_DELIMITER + 'start'
+}
+
+/// Return a string that encodes the given log level and name for a mark that indicates the end of
+/// an interval.
+function end_interval_label(log_level: LogLevel, interval_name: string): string {
+    return log_level.toString() + MESSAGE_DELIMITER + interval_name + MESSAGE_DELIMITER + 'end'
+}
+
+/// Return a string that encodes the given log level and name for a measurement.
+function measure_interval_label(log_level: LogLevel, interval_name: string): string {
+    return log_level.toString() + MESSAGE_DELIMITER + interval_name + MESSAGE_DELIMITER + 'measure'
+}
+
+function parse_log_level(log_level_name: string): LogLevel {
+    switch (log_level_name) {
+        case LogLevel.Section.toString():
+            return LogLevel.Section
+        case LogLevel.Task.toString():
+            return LogLevel.Task
+        case LogLevel.Detail.toString():
+            return LogLevel.Detail
+        case LogLevel.Debug.toString():
+            return LogLevel.Debug
+    }
+}
```

Comment:
a very repetitive code again. Please make the code without such blocks – adding or removing levels should be one-line change (or close to that).

Reply:
Refactored.